### DISCUSSION TITLE
How do you spy on a pydantic method in pytest?

### DISCUSSION THREAD

**Discussion by dazza-codes** [pydantic/pydantic#9837] [Question]
Has anyone figured out how to spy on a pydantic model instance?  According to this closed issue, it doesn't work and there is no fix from pytest-mock.

https://github.com/pytest-dev/pytest-mock/issues/363

The pytest-mock docs about [Spy](https://pytest-mock.readthedocs.io/en/stable/usage.html#spy) don't work with a pydantic model.

**Comment by sjdemartini** [comment]
The best workaround I've found is to spy/mock the _class_, rather than the instance. Not as ideal, but can get the job done in certain situations. 

For example:

```python
class MyModel(BaseModel):
    def my_method(self):
        pass


def test_my_model(mocker):
    my_model = MyModel()
    spy = mocker.spy(MyModel, "my_method")  # instead of `spy(my_model, "my_method")`
    my_model.my_method()
    spy.assert_called_once()
    spy.assert_called_with(my_model)
```
